ml-agents-envs/mlagents_envs/base_env.py

```diff
@@ -331,17 +345,16 @@ def close(self) -> None:
         """
         pass
 
+    @property
     @abstractmethod
-    def get_behavior_names(self) -> List[BehaviorName]:
+    def behavior_specs(self) -> BehaviorMapping:
```

Comment:
```suggestion
    def behavior_specs(self) -> Mapping:
```
I think the abstract class should just return an abstract Mapping. Other implementations could return a plain dict if they wanted.